code-challenges/regexp-part2/challenges-11.test.js

```diff
@@ -0,0 +1,165 @@
+'use strict';
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 1
+
+Write a function named validatePin that uses a regular expression pattern to validate a PIN.
+
+If the PIN is four numerical digits long, return true. Otherwise, return false.
+------------------------------------------------------------------------------------------------ */
+
+const validatePin = (pin) => {
+  // Solution code here...
+  let validator = /^[0-9]{4}$/
+  return validator.test(pin) ? true : false;
+
+};
+
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 2
+
+Write a function named validateEmail that takes in an email address and validates it based
+on several rules:
+  - one word, or two words separated by a period, before the @ symbol
+  - can contain numbers
+  - can have any of the following top-level domains: .net, .com, or .org
+  - no other special characters
+  - no subdomains, ports, etc: must be of the form name@place.com, not name@sub.place.com:3000
+
+Return either true or false.
+
+Note: if you ever need to validate an email using a regex in practice, the Internet has the actual regex you should use. It's many many lines long.
+------------------------------------------------------------------------------------------------ */
+
+const validateEmail = (email) => {
+  // Solution code here...
+  var validemail =/^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/;
+  return validemail.test(email) ? true :false;
```

Comment:
Where did you find this? Please give your sources credit!

Reply:
Sorry for not mentioning the source , 
https://stackoverflow.com/questions/4338267/validate-phone-number-with-javascript/4338544